Tests/Client/Unit Tests/AtomicTests.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  AtomicTests.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+@testable import StreamChatClient
+
+class AtomicTests: XCTestCase {
+
+    func test_Atomic_whenValueChanged_callsDidSetCallback() {
+        // Setup
+        let initialValue = "Luke"
+        let newValue = "Leia"
+
+        var callbackNewValue: String?
+        var callbackOldValue: String?
+
+        let value = Atomic(initialValue, callbackQueue: .testQueue) {
+            XCTAssertTrue(DispatchQueue.isTestQueue, "Callback called on an incorrect queue.")
+            callbackNewValue = $0
+            callbackOldValue = $1
+        }
+        assert(value.get() == initialValue)
+
+        // Action
+        value.set(newValue)
+
+        // Assert
+        AssertEqualEventually(callbackNewValue, newValue)
+        AssertEqualEventually(callbackOldValue, initialValue)
+    }
+
+    func test_Atomic_keyPathHelpers() {
+        // Setup
+        struct Helper: Equatable {
+            var elements: [Int] = []
+        }
+        let value = Atomic(Helper())
+
+        // Actions
+        value.update(\.elements, to: [0])
+        XCTAssertEqual(value.get(), Helper(elements: [0]))
+
+        value.elements = [1]
+        XCTAssertEqual(value.get(), Helper(elements: [1]))
+
+        value.append(to: \.elements, 2)
+        XCTAssertEqual(value.get(), Helper(elements: [1, 2]))
+    }
+
+    func test_Atomic_dictionaryHelpers() {
+        let value = Atomic(["Luke": 1])
+        XCTAssertEqual(value["Luke"], 1)
+    }
+
+    func test_Atomic_whenCalledRecursively_worksCorrectly() {
+        // Setup
+        let value = Atomic("Luke")
+
+        // Action
+        value.update { (oldValue) -> String? in
```

Comment:
In this test case, naming implies `value` is `String` or `oldValue` is Atomic. I suggest renaming `value` to `atomic` or `atomicValue`